// vice-react-playback/rollup.config.js
import babel from 'rollup-plugin-babel'
import resolve from '@rollup/plugin-node-resolve'
import external from 'rollup-plugin-peer-deps-external'
import ts from '@rollup/plugin-typescript'
// import typescript from 'rollup-plugin-typescript2'
import typescript from 'typescript'
import postcss from 'rollup-plugin-postcss'
import { terser } from 'rollup-plugin-terser'
import { DEFAULT_EXTENSIONS } from '@babel/core'

export default [
    {
        input: "./src/index.ts",
        output: [
            {
                file: 'dist/index.js',
                format: 'cjs',
                sourcemap: true
            },
            {
                file: 'dist/index.es.js',
                format: 'es',
                exports: 'named',
                sourcemap: true
            }
        ],
        plugins: [
            postcss({
                plugins: [],
                minimize: true
            }),
            babel({
                exclude: 'node_modules/**',
                presets: ['@babel/preset-react'],
                extensions: [
                    ...DEFAULT_EXTENSIONS, '.tsx', '.ts'
                ]
            }),
            external(),
            resolve(),
            ts({
                typescript,
                jsx: "react",
                esModuleInterop: true
                // tsconfig: './src/tsconfig.json'
            }),
            terser()
        ]
    },
]
